### DISCUSSION TITLE
how to update dictionary.json and restart spacecraft.js in realtime for changing parameters in realtime

### DISCUSSION THREAD

**Discussion by Vaibhav2121** [nasa/openmct#3806] [Community Q&A]
Dear Software desginers

I have an issue with this software, presently as shown in telemetry integration example parameters definition is defined in dictionary.json file. Suppose there are 1000 parameters and 10 users of software. Each user wants to monitor different parameters which he can change in realtime. How to modify dictionary.json and restart spacecraft.js in realtime to acquire new parameters in real time.

**Comment by DusterTheFirst** [comment]
The example application has the telemetry values hard-coded as an example. If you want to setup dynamic telemetry parameters, you can modify the node.js server code to accommodate that. This project is not a drop in application, but provides many APIs for a programmer to integrate it however they please. The code that you write will be specific to your application, but you can build off of the tutorial code, replacing the JSON file with something else that can be dynamically updated.

**Comment by DusterTheFirst** [comment]
This may be better suited as a discussion topic: https://github.com/nasa/openmct/discussions/categories/community-q-a

**Comment by jvigliotta** [comment]
@Vaibhav2121 @DusterTheFirst I'll move this over to discussions for any further... discussion.

**Comment by davetsay** [comment]
@davetsay

**Comment by davetsay** [comment]
@Vaibhav2121 , as @DusterTheFirst mentioned, you could set up dynamic telemetry parameters, which can change in realtime. You could also have multiple deployments with different telemetry parameters.

However, most spacecraft will have well defined, static, telemetry parameters. It sounds like this is an issue that Open MCT does very well, compose custom displays of visualized telemetry. I think your use case is solved not in setup of the application, but in using the application. With a static dictionary, users/operators still have the ability to pick and choose which telemetry parameters they wish to monitor, and which they do not want to monitor. Users can create views to support monitoring of different parameters. Display Layout, Flexible Layout, and Tabs, are a few of the creatable views in which telemetry can be combined to provide a comprehensive display of multiple parameters in various visualizations, such as plots or tables.

Some additional reading which may help in understanding how domain objects, views, and telemetry are connected.
https://github.com/nasa/openmct/blob/master/API.md#domain-objects-and-identifiers
https://github.com/nasa/openmct/blob/master/API.md#object-providers
https://github.com/nasa/openmct/blob/master/API.md#telemetry-api
